app/src/services/bucketPermission.js

```diff
@@ -108,6 +108,22 @@ const service = {
     }
   },
 
+  /**
+   * @function getBucketIdsWithObject
+   * Searches for specific (bucket) object permissions
+   * @param {string|string[]} [params.userId] Optional string or array of uuids representing the user
+   * @returns {Promise<object>} The result of running the find operation
+   */
+  getBucketIdsWithObject: async (userIds) => {
+    return ObjectPermission.query()
+      .select('bucketId')
+      .distinct('bucketId')
+      .joinRelated('object')
+      .whereIn('userId', userIds)
```

Comment:
Let's replace this line with `.modify('filterUserId', userIds)`. The way this is written right now, if you pass in a string to `userId`, it won't work. It only looks like it works because we already pre-cast it in the controller this is invoked in with the `mixedQueryToArray` function, which always makes everything an array, even if there's only one element. We should be able to support both a string, or an array of strings just like what the JSDocs specify on line 115.